### PR TITLE
replace mysql with mysqli and delete mysqli file fix #3774

### LINKED ISSUE #3774: Drop mysql_* ?
yoshi was so nice and place in his upgrade clean pr a logic which switch
the database driver from mysql to mysqli automaticaly.
i got now the question in my mind,
if we not drop the support for mysql_* stuff?

### PR DESCRIPTION
**Pull request by albertlast**
This pr based on the issue #3774 

@Yoshi2889 well you should than revoke you changes from https://github.com/SimpleMachines/SMF2.1/pull/3761/commits/49d6fe4729f7f38863b2c5d3f485c64547cc8991 sry

### CODE REVIEW

**Comment by NanoSector** [comment]
Or you can close this PR and wait for mine to be merged, or contribute to
my PR. The collaborative part of Github is completely gone if we are going
to copy each other's work and then get them to change their PR.

Op do 5 jan. 2017 13:38 schreef albertlast <notifications@github.com>:

**Comment by NanoSector** [comment]
Okay Github opened the wrong changeset. I can read the first 100 lines
before my browser crashes so I don't see how this is related to my code?

Op do 5 jan. 2017 13:41 schreef Rick Kerkhof <rick.2889@gmail.com>:

**Comment by live627** [comment]
Will test this later tonight.

File: Sources/Subs-Db-mysql.php [hunk 1]
```diff
@@ -7,7 +7,7 @@
  *
  * @package SMF
  * @author Simple Machines http://www.simplemachines.org
- * @copyright 2017 Simple Machines and individual contributors
+ * @copyright 2012 Simple Machines
```
Comment (Oldiesmann) [inline Sources/Subs-Db-mysql.php]: Please ensure you're using the latest code.

Comment (albertlast) [reply Sources/Subs-Db-mysql.php]: Please ensure that you update the copyright information in all files: https://github.com/SimpleMachines/SMF2.1/blob/release-2.1/Sources/Subs-Db-mysqli.php#L10
:)

File: Sources/Subs-Db-mysql.php [hunk 2]
```diff
@@ -33,44 +33,50 @@ function smf_db_initiate($db_server, $db_name, $db_user, $db_passwd, $db_prefix,
...
+			'db_sybase'                 => false,
+			'db_case_sensitive'         => false,
 			'db_escape_wildcard_string' => 'smf_db_escape_wildcard_string',
-			'db_is_resource' => 'is_resource',
+			'db_is_resource'            => 'smf_is_resource',
 		);
 
-	if (!empty($db_options['port']))
-		$db_server .= ':' . $db_options['port'];
-
```
Comment (Oldiesmann) [inline Sources/Subs-Db-mysql.php:64]: Please keep the above code in place, as it allows connecting on a non-standard port.

Comment (albertlast) [reply Sources/Subs-Db-mysql.php:64]: this file is one - one copy of the existing mysqli file this port stuff is done in line 73.
https://github.com/SimpleMachines/SMF2.1/blob/release-2.1/Sources/Subs-Db-mysqli.php#L73

File: Sources/Subs-Db-mysql.php (hunk 1 above)
Comment (Oldiesmann) [reply Sources/Subs-Db-mysql.php]: Whoops...

File: Sources/Subs-Db-mysql.php (hunk 2 above)
Comment (Oldiesmann) [reply Sources/Subs-Db-mysql.php:64]: Ah ok. Didn't see that :)

**Review by live627 — APPROVED** [review]

**Pull request merged**